src/azul/chalice.py

```diff
@@ -158,7 +158,7 @@ def self_url(self, endpoint_path=None) -> str:
             assert False, self.current_request
         if endpoint_path is None:
             endpoint_path = self.current_request.context['path']
-        return self_url.set(path=endpoint_path).url
+        return str(self_url.set(path=endpoint_path))
```

Comment:
Please create a FIXME to change `self_url` `file_url` and `manifest_url` to return a `furl` instance, and `get_manifest_async` to accept one. The overall goal is to avoid repeatedly converting `furl` instances to strings and vice versa.